### PR TITLE
Rename stripe-node to stripe

### LINKED ISSUE #12515: Stripe definitions are actually stripejs definitions
I tried using the latest `stripe/stripe.d.ts` file in this repo and had problems.
  - The authors of that type definition are cc/ @ahi @borisyankov @longzheng @ianschmitz @a904guy @amritk 

What we have there are in stripe definitions are actually types of the stripejs, not stripe node module itself. This lead to confusion and unusability when using stripe on the backend. Solution is either to rename stripe to stripejs (but this may be a problem if someone push a stripejs module with completely different implementation. Another solution is to keep it as it is and include backend stripe's typings into this package itself

### PR DESCRIPTION
**Pull request by ghost**
Fixes #12515.
The NPM package is named `stripe`, so it makes sense to name the types `@types/stripe` so that module resolution can work correctly.
If this is merged, I will run `npm deprecate @types/stripe-node "'@types/stripe-node' is now '@types/stripe'"`.
There was previously a published `@types/stripe` that was for the API definitions; those are now in `@types/stripe-v2` and `@types/stripe-v3`. Since the last published version of that was 3.0 and `stripe-node` is at 4.7, it should be fine to publish using that name.

CC @pleerock @bryanerayner @amritk @galuszkak @LinusU @SimonSchick

### CODE REVIEW

**Comment by andy-ms** [comment]
@RyanCavanaugh Is there any way to have both a call and construct signature and also be merged with a namespace?

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -8,17 +8,10 @@
 
 /// <reference types="node" />
 
-declare module 'stripe' {
-    var out: StripeNode.StripeExport;
-    export = out;
-}
+declare function StripeNode(apiKey: string, version?: string): StripeNode.Stripe;
+export = StripeNode;
```
Comment (LinusU) [inline types/stripe/index.d.ts]: Don't know if it matters, but they call this `stripePackage` in their readme...

[ref](https://github.com/stripe/stripe-node#usage)

```js
import stripePackage from 'stripe';
const stripe = stripePackage('sk_test_...');
```

Comment (LinusU) [reply types/stripe/index.d.ts]: Although, in the code it's named `Stripe` and it really seems like it's supposed to be a class, and not a function, even though their documentation kinda leads us on to think it's a function.

https://github.com/stripe/stripe-node/blob/a22a89c2e765d2548929d3d42dc0d7c4c4f6c19c/lib/stripe.js#L76-L109

Comment (LinusU) [reply types/stripe/index.d.ts]: This also would only be reflected in TypeScript if we declare it a class I think?

```js
const Stripe = require('stripe')
const stripe = new Stripe('...')

stripe instanceof Stripe
//=> true
```

Comment (LinusU) [reply types/stripe/index.d.ts]: The more I look the more I'm leaning against something like

```typescript
class Stripe {
  // ...
}

export = Stripe
```

**Comment by amritk** [comment]
The discussion here is worth reading:
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/12515

This change will break for people still using @typings/stripe for stripejs even though they should be using the versioned ones.

**Comment by RyanCavanaugh** [comment]
@andy-ms I can't think of a way to do that.

**Comment by andy-ms** [comment]
Fortunately it's a new major version publish, so shouldn't be a breaking change using the old `@types/stripe` which is at v3.

**Comment by amritk** [comment]
ok sweet

File: types/stripe/stripe-tests.ts [hunk 2]
```diff
@@ -1,11 +1,9 @@
 import StripeNode = require('stripe');
```
Comment (LinusU) [inline types/stripe/stripe-tests.ts]: potentially rename this to just `Stripe`

**Comment by LinusU** [comment]
Looks good to me 👍

**Comment by andy-ms** [comment]
Deprecated `@types/stripe-node`.

**Comment by SimonSchick** [comment]
@andy-ms is the move complete yet?

Right now `@types/stripe` appears to be a different set of typings.

**Comment by LinusU** [comment]
It just took a little while to get published, they should be live now 👍

**Pull request merged**